### PR TITLE
fix: do not crash on stopDaemon & upgrade to 0.4.3

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

**Comment by dignifiedquire** [comment]
cc @diasdavid

**Comment by dignifiedquire** [comment]
This can only be merged once https://github.com/ipfs/go-ipfs/issues/2748 is fixed and released as we can not use 0.4.2

**Comment by dignifiedquire** [comment]
Something seems off with go-ipfs-dep

```
ERR! argv "/home/ubuntu/nvm/versions/node/v6.7.0/bin/node" "/home/ubuntu/nvm/versions/node/v6.7.0/bin/npm" "install"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.3
npm ERR! path /home/ubuntu/js-ipfsd-ctl/node_modules/go-ipfs-dep/go-ipfs/ipfs
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall chmod

npm ERR! enoent ENOENT: no such file or directory, chmod '/home/ubuntu/js-ipfsd-ctl/node_modules/go-ipfs-dep/go-ipfs/ipfs'
npm ERR! enoent ENOENT: no such file or directory, chmod '/home/ubuntu/js-ipfsd-ctl/node_modules/go-ipfs-dep/go-ipfs/ipfs'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
```

@diasdavid any ideas?

File: src/node.js [hunk 1]
```diff
@@ -236,7 +257,7 @@ module.exports = class Node {
   setConfig (key, value, done) {
     run(this.exec, ['config', key, value, '--json'], {env: this.env})
       .on('error', done)
-      .on('data', (data) => {})
+      .on('data', () => {})
       .on('end', () => done())
```
Comment (diasdavid) [inline src/node.js]: I believe if this errors, it will call done twice, unless there is a test showing something different

**Comment by diasdavid** [comment]
@dignifiedquire can we get a description on what was the 'stopDaemon' bug?

File: src/node.js (hunk 1 above)
Comment (dignifiedquire) [inline src/node.js]: It's the same as it is currently on `master`. I can change it though.

**Comment by dignifiedquire** [comment]
> @dignifiedquire can we get a description on what was the 'stopDaemon' bug?

This is an issue I was experiencing in `master`, where calling `stopDaemon` would result in a crash, because it was still listening for error events.

File: src/node.js (hunk 1 above)
Comment (dignifiedquire) [inline src/node.js]: done

File: package.json [hunk 2]
```diff
@@ -44,18 +44,20 @@
   ],
   "license": "MIT",
   "dependencies": {
-    "go-ipfs-dep": "0.4.1",
+    "bl": "^1.1.2",
+    "go-ipfs-dep": "0.4.3",
```
Comment (diasdavid) [inline package.json]: You need to update to `0.4.3-1`, `0.4.3` has a bug that is only detected when it gets installed after publish (impossible to detect beforehand)

Comment (dignifiedquire) [inline package.json]: thanks <3

**Pull request merged**